### ISSUE TITLE
Grid View Info Box

### ISSUE THREAD

**Issue by AcroDirki** [OpenAero/main#207]
Hi guys!
I want to kindly ask you for the implementation of an info-box in the grid view. This box should contain counters, which give back the actual number of proposed figures with:
- postive entries
- negative entries
- positive exits
- negative exits

Background:
In several figure proposals of different glider aerobatic championships it was difficult for the competitors and the jury to immediately evaluate the feasibility of the proposed figures. Consider the following (more unlikely) example:
For 5 unknown sequences, 5x7 = 35 figures were proposed. Within this figure pool are 18 with negative exits, but only 2 with negative entries. Now, if you use all 5x2 Linking Figures appropriate and finish every sequence inverted (which is not allowed in the Advanced), there is 18-10-5-2 = 1 figure, which cannot match with the remaining figures without interruption. Hence, the last unknown sequence is not feasible by definition.
Practically, this problem didn't often occur, since either the weather solved this problem or the prescribed spread wasn't that large / unlikely.
However, such a box would be a useful assistant, probably even in powered aerobatics.

Further, especially in case of glider aerobatics, the entry and exit speeds are of interest for more or less the same reason. Imagine, there are 16 slow negative exits and 2 fast negative exits, but 13 fast negative entries. Then, at least 1 unharmonic speed adjustment would be required, which doesn't look professional.
The entry and exit speeds could either be counted by appropriate signs or signal words in the command line and/or probably by a graphical evaluation of the figure symbol (determination of figure line CG and relative distance to the entry and exit points). The identified speeds could be shown in the grid for manual checking.

Thank you for the attention and best regards,
D!

**Comment by AcroDirki** [comment]
In addition to these statistical information and in order to support the jury, it is desirable to invent a routine, which randomly assembles the unknown programmes based on the proposed figures and considering the prescribed number of entry and exit attitudes, the sequence total K and the linking/additional figures.
Imagine that for the last sequence remain 1 negative exit, but 5 negative entries.

**Comment by OpenAero** [comment]
The requested features have been added to the development version, except for manual editing of entry/exit speeds.
Please verify with relevant figure proposals on https://openaero.net/devel

**Comment by AcroDirki** [comment]
I've had a quick review so far. It worked excellent for some arbitrary proposed figures. I'll perform a more detailed check, using some entire figure proposals of the last championships. May be you can meanwhile explain briefly, how the identification / assignment of entry and exit speed categories works.

Again, great work!
Best regards

**Comment by OpenAero** [comment]
Hi Dirk,

The assignment is done as follows:
First of all, speeds are based on base figures. This keeps things simple
and means the speeds are only assigned during the start of OpenAero.
Of course this could be expanded to include rolls etc but this would make
things more complex and, in my opinion, more arbitrary.

The figure entry speed is defined by it's start:
Climbing = high speed
Descending = low speed
Neither = neutral

For the exit:
Descending = high speed
Climbing = low speed
Neither = neutral

Low and high speed can only be combined with the same type, or with neutral.

The resultant number of additional figures required certainly isn't
perfect. It can be too high or too low. But in general it does seem to give
a reasonable average.
An issue I have found myself is that this number can also depend on the
order in which the figures are presented. It should preferably not do this
and I am thinking about a way to solve this without creating many more
calculations.

Regards,
Ringo

Op maandag 7 november 2016 heeft AcroDirki <notifications@github.com> het
volgende geschreven:

**Comment by AcroDirki** [comment]
Hi Ringo
So, now I've had a closer look. 2, 3 issues:
 First, the box arrangement could be improved. On my display, the box overlaps with some proposed figures. Please, either put it on the left hand side below the figure editor, or may be in the bottom right corner of the display, or on the right side above the figure catalogue menu, or as the first or last grid position. Would be convenient to read the proposed figures and the grid info simultaneously, without any overlapping or scrolling.
Second, please provide a formula for the calculation for the number of additional figures in the manual, if possible. This makes it much more comprehensive to everybody. If its just an estimation rather than a determination, just declare it like this. It's just intended to be helpful - to reduce the concerns of the jury / contest director. In the end it remains to be their job to give corresponding advises and/or counter-measures. However, your finding above, I was unable to reproduce. I think it's more important for the user, how this result should be interpreted.
3rd, does the random algorithm accounts for additional figures? I mean i.e., is it possible that from 35 figures, which require 3 additionals, 7 figures are randomly selected, which still require 3 additionals, while there are a maximum of 2 additionals for a certain sequence? Then it would be inappropriate even if unlikely.

Best regards!

**Comment by OpenAero** [comment]
Hi Dirk,

I did consider all the positions you mention but found that there's always
the chance of hiding something useful. Maybe it would be best to allow the
whole panel to be dragged to any position? Then it can be positioned
whereever the user wants it while keeping it on top.

The calculation logic is rather complex and would take up quite a bit of
space in the manual. But maybe I could use something like 'Click here for
the full explanation'.

I have found a rather functional solution to the 'varying additionals
issue' (they could vary between 1 and 3 for the same figures) by also
arranging the figures in reverse.

The random algorithm works as follows:
1) create sets that do not require additionals, as long as possible (upto
maximum required length)
2) take the longest set
3) add the shortest sets until reaching required length (e.g. 7). Split if
necessary
4) repeat from 2)

In this way the additionals are fairly evenly distributed. For example,
assume we have 28 figure sets with the following lengths:
7, 6, 5, 5, 3, 1, 1

This would result in combinations:
7
6 + 1
5 + 1 + 1
5 + 2

Of course it's still up to the jury to use a combination that truly works.
This is just a method to generate possibilities quickly. It is NOT intended
that the first option generated is always used. That is why the button
'GENERATE NEW PROPOSALS' is right there on the dialog.

Regards,
Ringo


2016-11-07 20:56 GMT+01:00 AcroDirki <notifications@github.com>:
